src/changelog-config.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable node/no-process-env, node/no-sync */
+
+import fs from 'fs';
+import path from 'path';
+
+type PackageJson = {
+  autoChangelog: {
+    packageRename: {
+      originalLastestVersion: string;
+      originalTagPrefix: string;
+    };
+  };
+};
+
+/**
+ * Returns the latest version of the original package in case of package renamed.
+ *
+ * @returns The original package latest version.
+ */
+export function getOriginalLatestVersion(): string | null {
+  // Set automatically by Yarn 3.x
+  const packageJsonPath = process.env.npm_package_json;
+  if (packageJsonPath) {
+    const packageJson = path.resolve(packageJsonPath);
+    const packageJsonContent = JSON.parse(
+      fs.readFileSync(packageJson, 'utf8'),
+    ) as PackageJson;
+
+    if (
+      typeof packageJsonContent.autoChangelog?.packageRename
+        ?.originalLastestVersion === 'string'
+    ) {
+      return packageJsonContent.autoChangelog.packageRename
+        .originalLastestVersion;
+    }
+  }
+
+  return null;
+}
+
+/**
+ * Returns the original package tag prefix in case of package renamed.
+ *
+ * @returns The original package tag prefix.
+ */
+export function getOriginalTagPrefix(): string | null {
+  // Set automatically by Yarn 3.x
+  const packageJsonPath = process.env.npm_package_json;
+  if (packageJsonPath) {
+    const packageJson = path.resolve(packageJsonPath);
+    const packageJsonContent = JSON.parse(
+      fs.readFileSync(packageJson, 'utf8'),
+    ) as PackageJson;
+    if (
+      typeof packageJsonContent.autoChangelog?.packageRename
+        ?.originalTagPrefix === 'string'
+    ) {
+      return packageJsonContent.autoChangelog.packageRename.originalTagPrefix;
+    }
+  }
+
+  return null;
```

Comment:
A couple changes here could simplify the logic and add type safety:

```suggestion
  if (packageJsonPath === undefined) {
    return null;
  }

  const packageJson = path.resolve(packageJsonPath);
  const packageJsonContent: PackageJson = JSON.parse(
    fs.readFileSync(packageJson, 'utf8'),
  );

  const { originalTagPrefix } = packageJsonContent.autoChangelog?.packageRename ?? {}
  return typeof originalTagPrefix === 'string' ? originalTagPrefix : null
```

- Early exit in case of `undefined` `packageJsonPath` -> less nesting, explicit handling of edge case.
- Replacing the `as` type assertion for `packageJsonContent` with a type annotation. 
- Replacing long optional chaining expression for `typeof` check and return statement with `originalTagPrefix` variable.
  - Safely destructuring `originalTagPrefix` from possible `undefined` using `??` operator.

Reply:
1. Being a small function, both processing wise and readability, I don't see change can make much difference! but I think this check `if(packageJsonPath)` will better suit here compare to `packageJsonPath === undefined`. The first check will also take care in case of null.
2. Even if it's undefined, the === 'string' will take care. It'll return when the value is set for the property otherwise null

Reply:
- Future versions of node will enforce a `?: string | number | boolean` type for the `process.env` values, so I don't think a truthy vs. falsy check is safe here. 
- Currently, a `null` value assigned as a `process.env` will be implicitly converted to the string `"null"`, and future versions will throw an error for a `null` value, so a deep equals check for `undefined` seems sufficient.
- We might also want to put line 55 (`fs.readFileSync()`) inside of a try catch block to guard against invalid paths. 
- Destructuring `originalTagPrefix` isn't necessary, but it has the advantage of letting us avoid repeating the long optional chaining expression, especially if there's any chance of more code being added down the road that accesses the `originalTagPrefix` property.

Reply:
Resolving this conversation since this code was removed.